Lib/test/_test_eintr.py

```diff
@@ -152,6 +152,37 @@ def test_read(self):
                 self.assertEqual(data, os.read(rd, len(data)))
             self.assertEqual(proc.wait(), 0)
 
+    def test_readinto(self):
+        rd, wr = os.pipe()
+        self.addCleanup(os.close, rd)
+        # wr closed explicitly by parent
+
+        # the payload below are smaller than PIPE_BUF, hence the writes will be
+        # atomic
+        datas = [b"hello", b"world", b"spam"]
+        bufs = [bytearray(5), bytearray(5), bytearray(4)]
+
+        code = '\n'.join((
+            'import os, sys, time',
+            '',
+            'wr = int(sys.argv[1])',
+            'datas = %r' % datas,
+            'sleep_time = %r' % self.sleep_time,
+            '',
+            'for data in datas:',
+            '    # let the parent block on read()',
+            '    time.sleep(sleep_time)',
+            '    os.write(wr, data)',
+        ))
+
+        proc = self.subprocess(code, str(wr), pass_fds=[wr])
+        with kill_on_error(proc):
+            os.close(wr)
+            for data, buffer in zip(datas, bufs):
+                os.readinto(rd, buffer)
+                self.assertEqual(data, buffer)
```

Comment:
```suggestion
                self.assertEqual(buffer, datum)
```

Reply:
It should be "data" here, no?

Reply:
I should have made one suggestion instead of two. If we apply my previous suggestion (https://github.com/python/cpython/pull/129211#discussion_r1928895432), then it would be fine.

Reply:
Ah ok, I misread the previous suggestion.